app/assets/javascripts/modules/FirmMap.js

```diff
@@ -0,0 +1,144 @@
+ *
+ * @module FirmMap
+ * @returns {class} FirmMap
+ */
+
+define(['jquery', 'DoughBaseComponent'],
+       function($, DoughBaseComponent) {
+  'use strict';
+
+  var FirmMapProto,
+      defaultConfig = {
+        zoomLevel: 11
+      };
+
+  /**
+   * @constructor
+   * @extends {DoughBaseComponent}
+   * @param {HTMLElement} $el    Element with dough-component on it
+   * @param {Object}             config Hash of config options
+   */
+  function FirmMap($el, config) {
+    FirmMap.baseConstructor.call(this, $el, config, defaultConfig);
+  }
+
+  /**
+   * Inherit from base module, for shared methods and interface
+   */
+  DoughBaseComponent.extend(FirmMap);
+  FirmMap.componentName = 'FirmMap';
+  FirmMapProto = FirmMap.prototype;
+
+  /**
+   * Init function
+   *
+   * Set up listeners and accept promise
+   *
+   * @param {Object} initialised Promise passed from eventsWithPromises (RSVP Promise).
+   * @returns {MultiTableFilter}
+   */
+  FirmMapProto.init = function(initialised) {
+    this.initializeGoogleMaps();
+    this._initialisedSuccess(initialised);
+    return this;
+  };
+
+  /**
+   * initializeGoogleMaps
+   *
+   * Loads the google maps library and calls setupMap when it's done.
+   */
+  FirmMapProto.initializeGoogleMaps = function() {
+    var $map = this._getMapElement();
+    require(['async!http://maps.google.com/maps/api/js?key=' + this.config.apiKey],
+            $.proxy(this.setupMap, this));
+  };
+
+  /**
+   * setupMap
+   *
+   * Sets up the map and adds advisers to it.
+   * Is called by the google maps library.
+   */
+  FirmMapProto.setupMap = function() {
+    var $map = this._getMapElement(),
+        gMap = this._createMap($map);
+    this._positionMarkers(gMap);
+  };
+
+  /**
+   * _createMap
+   *
+   * Creates an instance of a google map, binding it to the param element.
+   *
+   * @private
+   * @param {HTMLElement} $map Element to contain map
+   * @return {GoogleMap} instance of google map
+   */
+  FirmMapProto._createMap = function($map) {
+    return new google.maps.Map($map[0], {
+      center: {
+        lat: this.config.center.lat,
+        lng: this.config.center.lng
+      },
+      zoom: this.config.zoomLevel
+    });
+  };
+
+  /**
+   * _positionMarkers
+   *
+   * Positions the adviser markers on the map.
+   *
+   * @private
+   * @param {GoogleMap} gMap Instance of the map to position markers onto
+   */
+  FirmMapProto._positionMarkers = function(gMap) {
+    var $advisers = this.$find('[data-dough-map-point]');
+
+    $advisers.each($.proxy(function(_, adviser) {
+      var $adviser = $(adviser),
+          markerConfig = this._generateMarkerConfig($adviser);
+
+      markerConfig.map = gMap;
+      new google.maps.Marker(markerConfig);
+    }, this));
+  };
+
+  /**
+   * _getMapElement
+   *
+   * @private
+   * @return {HTMLElement} Finds map element
+   */
+  FirmMapProto._getMapElement = function() {
+    return this.$find('[data-dough-map]');
```

Comment:
Agreed — it would be nice to have it really confined (at least in spirit) to the DOM tree of the component, rather than them all requiring jQuery and getting global access.